iset.mm

```diff
@@ -154327,6 +154363,119 @@ equivalent way to state real trichotomy (see further discussion at
+      NUXGOUVGUXKONUUCUUDWAWDVVBVUTUXNUVGUWGJDVNUXNNUVGUXTONUUEUUFWAWDWQWRWSWTG
+      MUVGVUHXAXBUYSVUJUYRUUOMBXCZXFBMUVGDUUOXEXDUYTUVGBVUAYTEUVGYTHZUYTUWRUWSV
+      VFVDVJONSSUUGVFWMUUMUVIUYSUUHXIXQXKUYTVUDUURUYTVUCUUQABUYRUYSAUUMUVIAUYEU
+      VFACUVHAUVHXMUVEAUVDABXNXOXPXRUYSAXLXRUYTUYFPZVUCUUPDIZNKZVVHUVMIZJKZUUQV
+      VGVUBVVHNUYTUYFVUJVUBVVHKUYSVUJUYRUYFVVEXSZBMUUNDUUOXTYAULVVGVVHSHZVVKVVI
+      YCVVGRSUUPDVUOVVGVUQWMVVGBRUUNUUOVVGBMRUUOVVLVUPVVGVURWMUUJUYTUYFYDYEZYEV
+      VKVVJUXMKZVVMVVIUXMJVVJUYBYBVVMUWSVVOVVIYCZVJSRUVMYHZVVMUWSPVVPUWMUWNVVQU
+      WOUWPSRUVMYGVCSRVVHNUVMYIYRYJYKWRVVGVVJUUPJVVGUWMUUPRHVVJUUPKUWOVVNRSUUPD
+      UUIXDULYMYNUNYOYPUUKUUL $.
+  $}
+
+  ${
+    $d A f x $.  $d V f x $.  $d a f x $.
+    $( Weak omniscience stated in terms of natural numbers.  Similar to
+       ~ iswomnimap but it will sometimes be more convenient to use ` 0 ` and
+       ` 1 ` rather than ` (/) ` and ` 1o ` .  (Contributed by Jim Kingdon,
+       20-Jun-2024.) $)
+    iswomninn $p |- ( A e. V -> ( A e. WOmni <-> A. f e. ( { 0 , 1 } ^m A )
+        DECID A. x e. A ( f ` x ) = 1 ) ) $=
+      ( va cz cv c1 caddc co cmpt cfrec wceq oveq1 cbvmptv freceq1 iswomninnlem
+      cc0 ax-mp ) ABCEFEGZHIJZKZRLZDUBAFAGZHIJZKZMUCUFRLMEAFUAUETUDHINORUBUFPSQ
+      $.
+  $}
+
+  ${
+    $d A i x $.  $d F i x $.  $d i ph x $.
+    redcwlpolemeq1.f $e |- ( ph -> F : NN --> { 0 , 1 } ) $.
+    redcwlpolemeq1.a $e |- A =
+      sum_ i e. NN ( ( 1 / ( 2 ^ i ) ) x. ( F ` i ) ) $.
+    $( Lemma for ~ redcwlpo .  A biconditionalized version of ~ trilpolemeq1 .
+       (Contributed by Jim Kingdon, 21-Jun-2024.) $)
+    redcwlpolemeq1 $p |- ( ph -> ( A = 1 <-> A. x e. NN ( F ` x ) = 1 ) ) $=
+      ( c1 wceq cv cfv cn wral wa simpr c2 co cmul csu wcel cc0 wf trilpolemeq1
+      cpr adantr cexp cdiv fveqeq2 simplr rspcdva oveq2d 2nn a1i nnexpcld nncnd
+      nnnn0d nnap0d recclapd mulid1d eqtrd sumeq2dv geoihalfsum 3eqtr4g impbida
+      eqcomi ) ACHIZBJZEKHIZBLMZAVFNBCDEALUAHUDEUBVFFUEGAVFOUCAVINZLHPDJZUFQZUG
+      QZVKEKZRQZDSLVMDSZCHVJLVOVMDVJVKLTZNZVOVMHRQVMVRVNHVMRVRVHVNHIBLVKVGVKHEU
+      HAVIVQUIVJVQOZUJUKVRVMVRVLVRVLVRPVKPLTVRULUMVRVKVSUPUNZUOVRVLVTUQURUSUTVA
+      GVPHDVBVEVCVD $.
+  $}
+
+  ${
+    $d f i j x y z $.
+    $( Decidability of real number equality implies the Weak Limited Principle
+       of Omniscience (WLPO).  We expect that we'd need some form of countable
+       choice to prove the converse.
+
+       Here's the outline of the proof.  Given an infinite sequence F of zeroes
+       and ones, we need to show the sequence is all ones or it is not.
+
+       Construct a real number A whose representation in base two consists of a
+       zero, a decimal point, and then the numbers of the sequence.  This real
+       number will equal one if and only if the sequence is all ones
+       ( ~ redcwlpolemeq1 ).  Because (by hypothesis) real number equality is
+       decidable, it is decidable whether the sequence is all ones.
```

Comment:
If I interpret this hypothesis correctly, it is not proven that real number equality is decidable (in iset.mm or in general?) yet, is it? That means that it is unclear if a theorem like ~ qdceq for rationals exists for reals, isn't it?
Maybe this should be documented somewhere...

Reply:
Ah thanks for saying something, that shows me that I didn't explain the point of these results very well.

The idea here is that LPO and WLPO are constructive taboos (or to put it less colorfully, that they cannot be derived from the axioms of IZF). I've pushed up another commit which adds some language to that effect and also re-words the sentence you are reacting to here (the use of "hypothesis" here was supposed to be akin to a proof which derives a contradiction but it was confusing wording, so I just reworked the whole sentence).

Not sure what you mean by "in general" as contrasted with iset.mm, but if you mean set.mm, all propositions are decidable there (including real number equality of course).